diff --git a/arch/arm64/boot/dts/vendor/qcom/alioth-sm8250.dtsi b/arch/arm64/boot/dts/vendor/qcom/alioth-sm8250.dtsi
index 8f203c9..619fda5 100644
--- a/arch/arm64/boot/dts/vendor/qcom/alioth-sm8250.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/alioth-sm8250.dtsi
@@ -521,8 +521,8 @@ And public attribution of xiaomi platforms(like alioth and so and)
 	qcom,sec-charger-config = <0>;
 	qcom,usb-icl-ua = <3000000>;
 	qcom,fcc-max-ua = <6000000>;
-	qcom,fv-max-uv = <4460000>;
-	qcom,non-fcc-fv-max-uv = <4450000>;
+	qcom,fv-max-uv = <4480000>;
+	qcom,non-fcc-fv-max-uv = <4470000>;
 	qcom,dc-icl-ua = <1000000>;
 	qcom,auto-recharge-soc = <99>;
 	qcom,chg-term-src = <1>;
@@ -536,10 +536,10 @@ And public attribution of xiaomi platforms(like alioth and so and)
 	qcom,battery-data = <&umi_batterydata>;
 	mi,six-pin-step-chg-params = <4250000  6000000
 				4450000  5400000
-				4460000  3536000>;
+				4480000  3536000>;
 	mi,six-pin-step-chg;
 	mi,six-pin-soc-th = <22>;
-	mi,fcc-batt-unverify-ua = <2000000>;
+	mi,fcc-batt-unverify-ua = <6000000>;
 	mi,support-ffc;
 	qcom,lpd-disable;
 	mi,use-bq-pump;
@@ -651,7 +651,7 @@ And public attribution of xiaomi platforms(like alioth and so and)
 	cp_qc30 {
 		compatible = "xiaomi,cp-qc30";
 		status = "ok";
-		mi,qc3-bat-volt-max = <4420>;
+		mi,qc3-bat-volt-max = <4440>;
 		mi,qc3-bat-curr-max = <3600>;
 		mi,qc3-bus-volt-max = <12000>;
 		mi,qc3-bus-curr-max = <2100>;
@@ -664,12 +664,12 @@ And public attribution of xiaomi platforms(like alioth and so and)
 	usbpd_pm {
 		compatible = "xiaomi,usbpd-pm";
 		status = "ok";
-		mi,pd-bat-volt-max = <4460>;
+		mi,pd-bat-volt-max = <4480>;
 		mi,pd-bat-curr-max = <6000>;
 		mi,pd-bus-volt-max = <12000>;
 		mi,pd-bus-curr-max = <3000>;
 		mi,pd-bus-curr-compensate  = <50>;
-		mi,pd-non-ffc-bat-volt-max = <4450>;
+		mi,pd-non-ffc-bat-volt-max = <4470>;
 		mi,pd-battery-warm-th = <480>;
 		mi,use-qcom-gauge;
 		//mi,cp-sec-enable;
