diff --git a/drivers/power/supply/qcom/qpnp-fg-gen4.c b/drivers/power/supply/qcom/qpnp-fg-gen4.c
index 78143ee..43cf070 100644
--- a/drivers/power/supply/qcom/qpnp-fg-gen4.c
+++ b/drivers/power/supply/qcom/qpnp-fg-gen4.c
@@ -7373,7 +7373,7 @@ static int fg_gen4_probe(struct platform_device *pdev)
 	chip->calib_level = -EINVAL;
 	fg->param.batt_soc = -EINVAL;
 	fg->fake_temp = -EINVAL;
-	fg->fake_authentic = -EINVAL;
+	fg->fake_authentic = 1;
 	fg->fake_chip_ok = -EINVAL;
 	fg->maxim_cycle_count = INT_MIN;
 	chip->vbat_critical_low_count = 0;
