--- a/include/linux/susfs_def.h
+++ b/include/linux/susfs_def.h
@@ -3,6 +3,19 @@
 
 #include <linux/bits.h>
 #include <linux/string.h>
+#include <linux/version.h> // We need check kernel version.
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0)
+#include <linux/cred.h>
+#endif
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 7, 0)
+#define d_in_lookup(dentry) (0)
+#define d_lookup_done(dentry) do {} while (0)
+#endif // to support 4.4 and older kernel
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 7, 0)
+#define GFP_KERNEL_ACCOUNT GFP_KERNEL
+#endif // to support 4.4 and older kernel
 
 /********/
 /* ENUM */
@@ -50,7 +63,7 @@
 #define FUSE_SUPER_MAGIC 0x65735546
 #endif
 /*
- * inode->i_mapping->flags => A 'unsigned long' type storing flag 'AS_FLAGS_', bit 1 to 31 is not usable since 6.12
+ * inode->i_state => A 'unsigned long' type storing flag 'AS_FLAGS_', bit 1 to 31 is not usable since 6.12
  * nd->state => storing flag 'ND_STATE_'
  * nd->flags => storing flag 'ND_FLAGS_'
  * task_struct->thread_info.flags => storing flag 'TIF_'
@@ -72,12 +85,13 @@
  
 #define MAGIC_MOUNT_WORKDIR "/debug_ramdisk/workdir"
 
+
 static inline bool susfs_starts_with(const char *str, const char *prefix) {
-    while (*prefix) {
-        if (*str++ != *prefix++)
-            return false;
-    }
-    return true;
+	while (*prefix) {
+		if (*str++ != *prefix++)
+			return false;
+	}
+	return true;
 }
 
 static inline bool susfs_ends_with(const char *str, const char *suffix) {
@@ -95,6 +109,46 @@
 	return !strcmp(str + str_len - suffix_len, suffix);
 }
 
+/* From KernelSU */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 2, 0)
+typedef const struct qstr *susfs_fname_t;
+#define susfs_fname_len(f) ((f)->len)
+#define susfs_fname_arg(f) ((f)->name)
+#else
+typedef const unsigned char *susfs_fname_t;
+#define susfs_fname_len(f) (strlen(f))
+#define susfs_fname_arg(f) (f)
+#endif
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
+#define SUSFS_DECL_FSNOTIFY_OPS(name)                                            \
+int name(struct fsnotify_mark *mark, u32 mask, struct inode *inode,    \
+struct inode *dir, const struct qstr *file_name, u32 cookie)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(5, 2, 0)
+#define SUSFS_DECL_FSNOTIFY_OPS(name)                                            \
+int name(struct fsnotify_group *group, struct inode *inode, u32 mask,  \
+const void *data, int data_type, susfs_fname_t file_name,       \
+u32 cookie, struct fsnotify_iter_info *iter_info)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 18, 0)
+#define SUSFS_DECL_FSNOTIFY_OPS(name)                                            \
+int name(struct fsnotify_group *group, struct inode *inode, u32 mask,  \
+const void *data, int data_type, susfs_fname_t file_name,       \
+u32 cookie, struct fsnotify_iter_info *iter_info)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0)
+#define SUSFS_DECL_FSNOTIFY_OPS(name)                                            \
+int name(struct fsnotify_group *group, struct inode *inode,            \
+struct fsnotify_mark *inode_mark,                             \
+struct fsnotify_mark *vfsmount_mark, u32 mask,                \
+const void *data, int data_type, susfs_fname_t file_name,       \
+u32 cookie, struct fsnotify_iter_info *iter_info)
+#else
+#define SUSFS_DECL_FSNOTIFY_OPS(name)                                            \
+int name(struct fsnotify_group *group, struct inode *inode,            \
+struct fsnotify_mark *inode_mark,                             \
+struct fsnotify_mark *vfsmount_mark, u32 mask, void *data,    \
+int data_type, susfs_fname_t file_name, u32 cookie)
+#endif
+
 static inline bool susfs_is_current_proc_umounted(void) {
 	return (likely(test_thread_flag(TIF_PROC_UMOUNTED)));
 }
@@ -121,7 +175,11 @@
 
 static inline bool susfs_is_current_proc_umounted_app(void) {
 	return (likely(test_thread_flag(TIF_PROC_UMOUNTED)) &&
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0)
+			__kuid_val(current_uid()) >= 10000);
+#else
 			current_uid().val >= 10000);
+#endif
 }
 
 static inline bool susfs_is_current_proc_no_su(void) {
@@ -138,15 +196,15 @@
 
 #define SUSFS_IS_INODE_SUS_MAP(inode) \
 		inode && inode->i_mapping && \
-		unlikely(test_bit(AS_FLAGS_SUS_MAP, &inode->i_mapping->flags)) && \
+		unlikely(test_bit(AS_FLAGS_SUS_MAP, &inode->i_state)) && \
 		susfs_is_current_proc_umounted_app()
 
 #define SUSFS_IS_INODE_OPEN_REDIRECT_WITHOUT_UID_CHECK(inode) \
 		inode && inode->i_mapping && \
-		unlikely(test_bit(AS_FLAGS_OPEN_REDIRECT, &inode->i_mapping->flags))
+		unlikely(test_bit(AS_FLAGS_OPEN_REDIRECT, &inode->i_state))
 
 #define SUSFS_IS_INODE_OPEN_REDIRECT(inode) \
 		inode && inode->i_mapping && \
-		unlikely(test_bit(AS_FLAGS_OPEN_REDIRECT, &inode->i_mapping->flags)) && \
+		unlikely(test_bit(AS_FLAGS_OPEN_REDIRECT, &inode->i_state)) && \
 		susfs_is_current_proc_umounted_app()
 #endif // #ifndef KSU_SUSFS_DEF_H
