diff --git a/kernel/sched_assist/sched_assist_slide_v1.c b/kernel/sched_assist/sched_assist_slide_v1.c
index 5b0313b..6074707 100644
--- a/kernel/sched_assist/sched_assist_slide_v1.c
+++ b/kernel/sched_assist/sched_assist_slide_v1.c
@@ -124,7 +124,7 @@ u64 _slide_get_boost_load(int cpu) {
 #define SCHED_CPUFREQ_BOOST (1U << 9)
 #endif
 #endif /* OPLUS_FEATURE_SCHED_ASSIST */
-void adjust_sched_assist_input_ctrl(){
+void adjust_sched_assist_input_ctrl(void){
 	if (!sysctl_input_boost_enabled)
 		return;
 	if(sysctl_slide_boost_enabled){
diff --git a/sound/soc/codecs/audio/sia81xx/sia81xx.c b/sound/soc/codecs/audio/sia81xx/sia81xx.c
index 9ca5471..cde69d9 100644
--- a/sound/soc/codecs/audio/sia81xx/sia81xx.c
+++ b/sound/soc/codecs/audio/sia81xx/sia81xx.c
@@ -1650,11 +1650,11 @@ static int sia81xx_volme_boost_set(
 	return 0;
 }
 
-void sia81xx_start(){
+void sia81xx_start(void){
         sia81xx_resume(g_sia81xx);
 }
 
-void sia81xx_stop(){
+void sia81xx_stop(void){
         sia81xx_suspend(g_sia81xx);
 }
 #endif /* OPLUS_BUG_COMPATIBILITY */
