diff --git a/kernel/signal.c b/kernel/signal.c
index 4d0f34170..ab0d9d95b 100644
--- a/kernel/signal.c
+++ b/kernel/signal.c
@@ -28,6 +28,9 @@
 #include <linux/ratelimit.h>
 #include <linux/tracehook.h>
 #include <linux/capability.h>
+#ifdef CONFIG_REKERNEL
+#include "../drivers/rekernel/rekernel.h"
+#endif /* CONFIG_REKERNEL */
 #include <linux/freezer.h>
 #include <linux/pid_namespace.h>
 #include <linux/nsproxy.h>
@@ -1207,6 +1210,11 @@ int do_send_sig_info(int sig, struct siginfo *info, struct task_struct *p,
 	unsigned long flags;
 	int ret = -ESRCH;
 
+#ifdef CONFIG_REKERNEL
+	if (p && (sig == SIGKILL || sig == SIGTERM || sig == SIGABRT || sig == SIGQUIT))
+		rekernel_report(SIGNAL, sig, current->pid, current,
+				task_pid_nr(p), p, 0, NULL);
+#endif /* CONFIG_REKERNEL */
 	if (lock_task_sighand(p, &flags)) {
 		ret = send_signal(sig, info, p, group);
 		unlock_task_sighand(p, &flags);
