diff --git a/drivers/Kconfig b/drivers/Kconfig
index 4ee7416ed..574be56d9 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -216,4 +216,5 @@ source "drivers/gps/Kconfig"
 
 source "drivers/halls/Kconfig"
 
+source "drivers/rekernel/Kconfig"
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 8daadc6db..4e0bcc899 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -183,3 +183,4 @@ obj-$(CONFIG_SENSORS_SSC)		+= sensors/
 obj-$(CONFIG_TEE)		+= tee/
 obj-$(CONFIG_BCM_GPS_SPI_DRIVER) += gps/
 obj-$(CONFIG_HALLS)		+= halls/
+obj-$(CONFIG_REKERNEL) += rekernel/
